### ISSUE TITLE
Wired Loss Curve

### ISSUE THREAD

**Issue by Zihang-Xu-2002** [OptimalScale/LMFlow#831]
I trained the llama3 on my own conversation dataset with the command : 
./scripts/run_finetune.sh \
  --model_name_or_path meta-llama/Meta-Llama-3-8B \
  --dataset_path data/alpaca_selected/train \
  --conversation_template llama3 \
  --output_model_path output_models/finetuned_llama3_8b_selected

The initial learning rate is 2e-5 and batchsize_per_device is 4
And I found there are sharp drops at the beginning of every epoch. But during the epoch, there's no obvious loss drop.
<img width="455" alt="image" src="https://github.com/OptimalScale/LMFlow/assets/81083353/9202f59e-66e0-4ccc-bb7c-a4c349f42d06">

Before this I trained llama2
./scripts/run_finetune.sh \
  --model_name_or_path meta-llama/Llama-2-7b-hf \
  --dataset_path data/alpaca_raw/train \
  --conversation_template llama2 \
  --output_model_path output_models/finetuned_llama2_7b_raw

The initial learning rate is 8e-6 and batchsize_per_device is 4. The loss looks like :
<img width="443" alt="image" src="https://github.com/OptimalScale/LMFlow/assets/81083353/e16f43c1-bbb4-46e3-96fa-663c07cf08b4">

I am not sure if the gradient accumulation leads to this. I modified the "gradient_accumulation_steps" in configs/ds_config_zero3.json to 1 . But there's no changes.
<img width="316" alt="image" src="https://github.com/OptimalScale/LMFlow/assets/81083353/b0a9c346-78ec-4b39-93b3-69bc63da8d1b">

Could you help me with this issue? Thank you for your time and attention.

**Comment by research4pan** [comment]
Thanks for your interest in LMFlow! We've observed similar loss curves in some of our experiments. After careful examination, we attribute this to the overfitting of instruction following dataset on llama models. You may check your evaluation/test results, if the results are normal then it may not be a serious issue 😄
